src/status_im2/navigation/events.cljs

```diff
@@ -27,13 +28,13 @@
   [_ comp-id]
   {:navigate-to-within-stack comp-id})
 
-(rf/defn open-modal
-  {:events [:open-modal]}
-  [{:keys [db]} comp screen-params]
-  {:db            (-> (assoc db :view-id comp)
-                      (all-screens-params comp screen-params))
-   :dispatch      [:hide-bottom-sheet]
-   :open-modal-fx comp})
+(re-frame/reg-event-fx :open-modal
```

Comment:
this is a good example that we should consider, we don't want to know that each time we want to open modal we need to hide a sheet, so for these cases, we should have a function something like
(defn open-modal []
   (dispatch [open-modal])
   (dispatch [hide-bottom-sheet])

and then `on-press navigation/open-modal`

yeah but i feel it won't work for all cases, for example what if we have a condition and we want to open modal not always, btw in that case we'll have something like 

```clojure
:fx [[:dispatch [:hide-bottom-sheet]]
        (when something? [:open-modal-fx component])]
         ```

which is also not so great

Reply:
I think we're complicating a bit. To me it's just a matter of revisiting the event and make it more useful, accept an argument saying auto hide the bottom sheet, and so on.